service_container.rst

```diff
@@ -133,68 +133,61 @@ the service container *how* to instantiate it:
 
         # app/config/services.yml
         services:
-            app.message_generator:
-                class:     AppBundle\Service\MessageGenerator
-                arguments: []
+            # configures defaults for all services in this file
+            _defaults:
+                autowire: true
+                autoconfigure: true
+
+            # registers all classes in the dir(s) as services
+            AppBundle\:
+                resource: '../../src/AppBundle/{Service}'
 
     .. code-block:: xml
 
         <!-- app/config/services.xml -->
-        <?xml version="1.0" encoding="UTF-8" ?>
-        <container xmlns="http://symfony.com/schema/dic/services"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://symfony.com/schema/dic/services
-                http://symfony.com/schema/dic/services/services-1.0.xsd">
-
-            <services>
-                <service id="app.message_generator" class="AppBundle\Service\MessageGenerator">
-                </service>
-            </services>
-        </container>
+        TODO
 
     .. code-block:: php
 
         // app/config/services.php
-        use AppBundle\Service\MessageGenerator;
-        use Symfony\Component\DependencyInjection\Definition;
+        TODO
 
-        $container->setDefinition('app.message_generator', new Definition(
-            MessageGenerator::class,
-            array()
-        ));
+That's it! Thanks to the ``AppBundle\`` line and ``resource`` key below it, all
+classes in the ``src/AppBundle/Service`` directory will automatically be added to
+the container. Each service's "key" is simply its class name. You can use it immediately
+inside your controller::
```

Comment:
should we encourage type hinting arguments on the action method instead of using "get"?
fading out container->get provides many benefits, both at the design level and at the perf level (container optims of private services)